Source/Core/DiscIO/DirectoryBlob.h

```diff
@@ -170,10 +173,9 @@ class DiscContent
 class DiscContentContainer
 {
 public:
-  template <typename T>
-  void AddReference(u64 offset, const std::vector<T>& vector)
+  void Add(u64 offset, std::vector<u8> vector)
   {
-    return Add(offset, vector.size() * sizeof(T), reinterpret_cast<const u8*>(vector.data()));
+    return Add(offset, vector.size(), std::make_shared<std::vector<u8>>(std::move(vector)));
```

Comment:
It's important that `vector.size()` gets called before `std::move`, because otherwise the result of `vector.size()` will be 0. C++ doesn't guarantee the order of evaluation for function arguments, so you'll have to do it by assigning the result of `vector.size()` to a variable on a separate line.